cpp/mcap/include/mcap/reader.inl

```diff
@@ -1251,10 +1259,24 @@ TypedChunkReader::TypedChunkReader()
     , status_{StatusCode::Success} {}
 
 void TypedChunkReader::reset(const Chunk& chunk, Compression compression) {
-  ICompressedReader* decompressor =
-    (compression == Compression::None)  ? static_cast<ICompressedReader*>(&uncompressedReader_)
-    : (compression == Compression::Lz4) ? static_cast<ICompressedReader*>(&lz4Reader_)
-                                        : static_cast<ICompressedReader*>(&zstdReader_);
+  ICompressedReader* decompressor;
+
+  switch (compression) {
+    case Compression::None:
+    default:
+      decompressor = static_cast<ICompressedReader*>(&uncompressedReader_);
+      break;
+#ifndef MCAP_COMPRESSION_NO_LZ4
+    case Compression::Lz4:
+      decompressor = static_cast<ICompressedReader*>(&lz4Reader_);
+      break;
+#endif
+#ifndef MCAP_COMPRESSION_NO_ZSTD
+    case Compression::Zstd:
+      decompressor = static_cast<ICompressedReader*>(&zstdReader_);
+      break;
```

Comment:
An MCAP chunk containing an unrecognized compression string will cause a reader to defereference a null pointer 
here. Before this patch we would have assumed any unrecognized encoding was `zstd`, which would have caused some confusing error message, but likely not a segfault.

Would you be able to add an early return here in case of an unrecognized compression method?

Reply:
You mean `decompressor` pointer? It is initialized to uncompressed reader in the default switch case above.

Reply:
I've modified it slightly to make behavior the same as before when ZSTD is enabled.

Reply:
I don't think it makes sense to default to zstd here, we should set `status_` and return.

When both lz4 and zstd are built into the reader, this switch case is exhaustive. This is why the existing code works: we detected unrecognised compression strings in `ParseCompression` before ever getting here.

We're introducing a new error condition here for when a compression string is recognised but not supported. I think it makes sense to add a new error in `errors.hpp` for this. We could then go through all the supported compression values in this switch statement, and in the `default:` case set `status_ = ` and return early.

Reply:
Ok, now ParseCompression should fail if compression is not compiled in.